crypto/src/main/java/org/hyperledger/besu/crypto/LibSecp256k1.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.crypto;
+
+import java.nio.ByteBuffer;
+
+import com.sun.jna.Callback;
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.Structure;
+import com.sun.jna.Structure.FieldOrder;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+import com.sun.jna.ptr.PointerByReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public interface LibSecp256k1 extends Library {
+
+  /* Flags to pass to secp256k1_context_create */
+  public static final int SECP256K1_CONTEXT_VERIFY = 0x0101;
+  public static final int SECP256K1_CONTEXT_SIGN = 0x0201;
+
+  /* Flag to pass to secp256k1_ec_pubkey_serialize. */
+  public static final int SECP256K1_EC_UNCOMPRESSED = 0x0002;
+
+  static final LibSecp256k1 INSTANCE = findInstance();
+  static final PointerByReference CONTEXT = createContext();
+
+  private static LibSecp256k1 findInstance() {
```

Comment:
Maybe an `info` or `debug` log statement in `findInstance()` would make sense, to let the user know that either native is or is not being used?